apps/studio/pages/project/[ref]/storage/files/buckets/[bucketId].tsx

```diff
@@ -88,11 +130,25 @@ const BucketPage: NextPageWithLayout = () => {
         </div>
       </PageLayout>
 
-      <EditBucketModal
-        visible={showEditModal}
-        bucket={bucket}
-        onClose={() => setShowEditModal(false)}
-      />
+      {bucket && (
+        <>
+          <EditBucketModal
+            visible={modal === 'edit'}
+            bucket={bucket}
+            onClose={() => setModal(null)}
+          />
+          <EmptyBucketModal
+            visible={modal === 'empty'}
+            bucket={bucket}
+            onClose={() => setModal(null)}
+          />
+          <DeleteBucketModal
+            visible={modal === 'delete'}
+            bucket={bucket}
+            onClose={() => setModal(null)}
+          />
+        </>
+      )}
```

Comment:
These were removed from the root Files page so now added here, within the bucket page itself (via a dropdown button).